rsec16/coder.go

```diff
@@ -205,7 +211,8 @@ func (c Coder) ReconstructData(data, parity [][]byte) error {
 	}
 
 	if len(input) < c.dataShards {
-		return errors.New("not enough parity shards")
+		//not enough parity shards for repair
+		return IrrepairableError{}
```

Comment:
Hmm, it's a shame that the returned error type here is somewhat generic -- it seems to me that only the main function should really be the one figuring out what's a repairable error and what's not, especially since we don't want a reconstruction matrix failure (see comment below) to return a file I/O error.

Can you name the error type something like `NotEnoughParityShardsError`?